drivers/i2c/i2c_ite_enhance.c

```diff
@@ -637,77 +623,74 @@ static int i2c_enhance_init(const struct device *dev)
 		return error;
 	}
 
-	/* The pin is set to I2C alternate function of SCL */
-	pinmux_pin_set(config->alts_list[SCL].pinctrls,
-		       config->alts_list[SCL].pin,
-		       config->alts_list[SCL].alt_fun);
-	/* The pin is set to I2C alternate function of SDA */
-	pinmux_pin_set(config->alts_list[SDA].pinctrls,
-		       config->alts_list[SDA].pin,
-		       config->alts_list[SDA].alt_fun);
+	/* Set the pin to I2C alternate function. */
+	status = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (status < 0) {
+		LOG_ERR("Failed to configure I2C pins");
+		return status;
+	}
+
 
 	return 0;
 }
 
 static int i2c_enhance_recover_bus(const struct device *dev)
 {
 	const struct i2c_enhance_config *config = dev->config;
-	int i;
+	int i, status;
+	uint8_t scl_pin, sda_pin;
+
+	scl_pin = config->pcfg->states->pins[SCL].pin;
+	sda_pin = config->pcfg->states->pins[SDA].pin;
```

Comment:
pinctrl configuration is considered opaque (and subject to change at any time), so drivers can't access this information. Please find an alternative here. You may have to duplicate pin information for this particular case, since you need to make use of the GPIO API e.g. `sda-gpios = <&gpioa 0 ...>;`. You'll likely need a custom "gpio" pinctrl state so that the pinctrl drivers configures the pins back to "gpio" mode as well.

Reply:
Done.